hir2/src/dialects/builtin/builders/component.rs

```diff
@@ -34,6 +35,25 @@ impl ComponentBuilder {
 
     pub fn define_module(&mut self, name: Ident) -> Result<ModuleRef, Report> {
         let builder = PrimModuleBuilder::new(&mut self.builder, name.span());
-        builder(name)
+        let module_ref = builder(name)?;
+        let is_new = self
+            .component
+            .borrow_mut()
+            .symbol_manager_mut()
+            .insert_new(module_ref, crate::ProgramPoint::Invalid);
```

Comment:
@bitwalker I'm wondering if the symbol registration is better to be done in the op builder (module in this case)? When I first encountered this (function creation in the module) I was quite surprised that new function is not automatically registered in module symbol table.

Reply:
Good catch, I must've never added the logic to automatically register new symbol ops when inserted into a symbol table op. That should probably be done as some sort of hook that fires anytime an op is inserted into a region (or removed from one), but for now we'll have to just implement the logic in the op-specific builders manually.